src/wallet/spend.cpp

```diff
@@ -225,17 +223,13 @@ CoinsResult AvailableCoins(const CWallet& wallet,
     return result;
 }
 
-CAmount GetAvailableBalance(const CWallet& wallet, const CCoinControl* coinControl)
+CAmount GetAvailableBalance(const CWallet& wallet, const CCoinControl* _coinControl)
 {
     LOCK(wallet.cs_wallet);
-
-    CAmount balance = 0;
-    for (const COutput& out : AvailableCoins(wallet, coinControl).coins) {
-        if (out.spendable) {
-            balance += out.txout.nValue;
-        }
-    }
-    return balance;
+    // Copy to filter by spendable coins only
+    CCoinControl coinControl = _coinControl ? *_coinControl : CCoinControl();
```

Comment:
In 2cdd2a1a5498cb8767799dd26a4082e9d3b064ee "wallet: speedup 'GetAvailableBalance' filtering by spendable outputs only and using the 'AvailableCoins' retrieved total_amount."

What is the purpose of the rename and making an empty `CCoinControl` here? `AvailableCoins` can take `nullptr` for `coinControl`.

Reply:
The previous flow was:
```c++
std::vector<COutput> vCoins;
AvailableCoins(wallet, vCoins, coinControl);
for (const COutput& out : vCoins) {
     if (out.spendable) { --> // this is the reason
         balance += out.txout.nValue;
     }
}
```

To remove the second unnecessary for-loop, as the function arg is `const CCoinControl*`, in order to continue having the same calculation (skipping the non-spendable coins), needed to create a local copy of the `coinControl` arg and set the new flag `m_include_only_spendable_outputs` to true (which by default is false).

So the coins vector and the total balance returned by `AvailableCoins ` has all, and only, the spendable coins directly.